src/DotLiquid.Tests/StandardFilterTests.cs

```diff
@@ -1205,6 +1205,31 @@ public void TestPlusStringV21()
             TestPlus(context);
         }
 
+        [Test]
+        public void TestPlusNumericBoundary()
+        {
+            TestPlusNumericBoundaryByContext(_contextV20);
+            TestPlusNumericBoundaryByContext(_contextV21);
+            Helper.AssertTemplateResult(expected: "2147483648", template: "{{ '2147483647' | plus: 1 }}", syntax: _contextV21.SyntaxCompatibilityLevel);
+        }
+
+        private void TestPlusNumericBoundaryByContext(Context context)
+        {
+            // NOTE(David Burg): Tests promotion from 32-bit to 64-bit integer.
+            Helper.AssertTemplateResult(
+                expected: "2147483648",
+                template: "{{ big | plus: 1 }}",
+                localVariables: Hash.FromAnonymousObject(new { big = int.MaxValue }),
+                syntax: context.SyntaxCompatibilityLevel);
+
+            // NOTE(David Burg): We don't have support for BigInteger as this time.
+            Helper.AssertTemplateResult(

```

Comment:
todo: also add tests demonstrating that byte max, short max work already, but double range limit cannot be extended beyond current overflow.